tools/floodgate/js/floodgate.js

```diff
@@ -40,8 +42,17 @@ async function refreshPage(config, projectDetail, project) {
 
 function setListeners(project, projectDetail) {
   document.querySelector('#reloadProject button').addEventListener('click', reloadProject);
-  document.querySelector('#copyFiles button').addEventListener('click', () => floodgateContent(project, projectDetail));
-  document.querySelector('#promoteFiles button').addEventListener('click', () => promoteFloodgatedFiles(project));
+  document.querySelector('#copyFiles button').addEventListener('click', (e) => {
+    modal.getElementsByTagName('p')[0].innerText = `Confirm to ${e.target.textContent}`;
+    document.querySelector('#fg-modal #yes-btn').addEventListener('click', () => { modal.style.display = 'none'; floodgateContent(project, projectDetail); });
+    modal.style.display = 'block';
+  });
```

Comment:
I don't love that you're setting an event listener within another listener. Is there a reason why you can't set it outside?